### PR TITLE
DocumentCard: Convert to MergeStyles

### LINKED ISSUE #5408: DocumentCard: Convert to CSS-in-JS
**Describe the feature that you would like added**
This feature is to convert DocumentCard as a whole to use mergestyles to take advantage of theming and styling.

### PR DESCRIPTION
**Pull request by mikewheaton**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #5408
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Updates DocumentCard (and all of its sub-components) to use MergeStyles.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.styles.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        padding: `${verticalPadding}px ${horizontalPadding}px`,
+        position: 'relative'
+      },
+      className
+    ],
+    action: [
+      classNames.action,
+      {
+        float: 'left',
+        marginRight: '4px',
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.styles.ts]: nit: any rule that receives just a px value can be left as the number only. merge-styles will convert it automatically to px. You might save some bytes if fixing it everywhere 😄

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.styles.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        color: palette.neutralSecondary,
+        cursor: 'pointer',
+
+        selectors: {
+          '.ms-Button': {
+            fontSize: '16px',
+            height: `${actionSize}px`,
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.styles.ts]: same here... no need for string interpolation. you can leave it `actionSize` in here.

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.styles.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            cursor: 'pointer'
+          }
+        }
+      }
+    ],
+    views: [
+      classNames.views,
+      {
+        textAlign: 'right',
+        lineHeight: actionSize
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.styles.ts]: like here... the same thing can be done on line 41))

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.types.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import { IButtonProps } from '../../Button';
+import { IStyle, ITheme } from '../../Styling';
+import { IRefObject, IStyleFunctionOrObject } from '../../Utilities';
+import { DocumentCardActionsBase } from './DocumentCardActions.base';
+
+export interface IDocumentCardActionsProps extends React.ClassAttributes<DocumentCardActionsBase> {
+  /**
+   * Gets the component ref.
+   */
+  componentRef?: IRefObject<{}>;
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.types.ts]: nit: Not sure about it but better to have an empty interface of `IDocumentCardActions` and pass it to `IRefObject` in here. There are many components following this pattern. This way you have a type-safe hook in the future if you want to add some publicly accessible methods on the instance of the component.

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.styles.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
+import { getGlobalClassNames, FontSizes, FontWeights } from '../../Styling';
+import { IDocumentCardActivityStyleProps, IDocumentCardActivityStyles } from './DocumentCardActivity.types';
+
+const verticalPadding = 8;
+const horizontalPadding = 16;
+const imageSize = 32;
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.styles.ts]: Shouldn't all these consts be capitalized? `IMAGE_SIZE` vs `imageSize`

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.styles.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            }
+          }
+        }
+      ],
+      compact && [
+        classNames.rootCompact,
+        {
+          display: 'flex',
+          maxWidth: '480px',
+          height: '109px',
```
[the remark this answers is not in the document]
Comment (Vitalius1) [reply packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.styles.ts]: just curious here 😄

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.tsx [hunk 7]
```diff
@@ -1,33 +1,11 @@
...
-        )}
-      </div>
-    );
-  }
-}
+export const DocumentCardActions = styled<IDocumentCardActionsProps, IDocumentCardActionsStyleProps, IDocumentCardActionsStyles>(
+  DocumentCardActionsBase,
+  getStyles,
+  undefined,
+  { scope: 'DocumentCardActionsBase' }
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.tsx]: Scope should be `DocumentCardActions`

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.tsx [hunk 8]
```diff
@@ -1,66 +1,11 @@
...
+import { styled } from '../../Utilities';
+import { DocumentCardActivityBase } from './DocumentCardActivity.base';
+import { getStyles } from './DocumentCardActivity.styles';
+import { IDocumentCardActivityProps, IDocumentCardActivityStyleProps, IDocumentCardActivityStyles } from './DocumentCardActivity.types';
+
+export const DocumentCardActivity = styled<IDocumentCardActivityProps, IDocumentCardActivityStyleProps, IDocumentCardActivityStyles>(
+  DocumentCardActivityBase,
+  getStyles,
+  undefined,
+  { scope: 'DocumentCardActivityBase' }
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.tsx]: Scope should be `DocumentCardActivity`

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardDetails.tsx [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+import { styled } from '../../Utilities';
+import { DocumentCardDetailsBase } from './DocumentCardDetails.base';
+import { getStyles } from './DocumentCardDetails.styles';
+import { IDocumentCardDetailsProps, IDocumentCardDetailsStyleProps, IDocumentCardDetailsStyles } from './DocumentCardDetails.types';
+
+export const DocumentCardDetails = styled<IDocumentCardDetailsProps, IDocumentCardDetailsStyleProps, IDocumentCardDetailsStyles>(
+  DocumentCardDetailsBase,
+  getStyles,
+  undefined,
+  { scope: 'DocumentCardDetailsBase' }
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardDetails.tsx]: Scope should be `DocumentCardDetails`

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardLocation.tsx [hunk 10]
```diff
@@ -1,17 +1,11 @@
...
-        {location}
-      </a>
-    );
-  }
-}
+export const DocumentCardLocation = styled<IDocumentCardLocationProps, IDocumentCardLocationStyleProps, IDocumentCardLocationStyles>(
+  DocumentCardLocationBase,
+  getStyles,
+  undefined,
+  { scope: 'DocumentCardLocationBase' }
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardLocation.tsx]: Scope should be `DocumentCardLocation`

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardLogo.tsx [hunk 11]
```diff
@@ -1,18 +1,11 @@
...
-        <Icon iconName={logoIcon} />
-      </div>
-    );
-  }
-}
+export const DocumentCardLogo = styled<IDocumentCardLogoProps, IDocumentCardLogoStyleProps, IDocumentCardLogoStyles>(
+  DocumentCardLogoBase,
+  getStyles,
+  undefined,
+  { scope: 'DocumentCardLogoBase' }
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardLogo.tsx]: Scope should be `DocumentCardLogo`

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardPreview.tsx [hunk 12]
```diff
@@ -1,103 +1,11 @@
...
+import { styled } from '../../Utilities';
+import { DocumentCardPreviewBase } from './DocumentCardPreview.base';
+import { getStyles } from './DocumentCardPreview.styles';
+import { IDocumentCardPreviewProps, IDocumentCardPreviewStyleProps, IDocumentCardPreviewStyles } from './DocumentCardPreview.types';
+
+export const DocumentCardPreview = styled<IDocumentCardPreviewProps, IDocumentCardPreviewStyleProps, IDocumentCardPreviewStyles>(
+  DocumentCardPreviewBase,
+  getStyles,
+  undefined,
+  { scope: 'DocumentCardPreviewBase' }
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardPreview.tsx]: Scope should be `DocumentCardPreview`

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardStatus.tsx [hunk 13]
```diff
@@ -1,29 +1,11 @@
...
-        {status}
-      </div>
-    );
-  }
-}
+export const DocumentCardStatus = styled<IDocumentCardStatusProps, IDocumentCardStatusStyleProps, IDocumentCardStatusStyles>(
+  DocumentCardStatusBase,
+  getStyles,
+  undefined,
+  { scope: 'DocumentCardStatusBase' }
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardStatus.tsx]: Scope should be `DocumentCardStatus`

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardTitle.tsx [hunk 14]
```diff
@@ -1,167 +1,11 @@
...
+import { styled } from '../../Utilities';
+import { DocumentCardTitleBase } from './DocumentCardTitle.base';
+import { getStyles } from './DocumentCardTitle.styles';
+import { IDocumentCardTitleProps, IDocumentCardTitleStyleProps, IDocumentCardTitleStyles } from './DocumentCardTitle.types';
+
+export const DocumentCardTitle = styled<IDocumentCardTitleProps, IDocumentCardTitleStyleProps, IDocumentCardTitleStyles>(
+  DocumentCardTitleBase,
+  getStyles,
+  undefined,
+  { scope: 'DocumentCardTitleBase' }
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardTitle.tsx]: Scope should be `DocumentCardTitle`

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.styles.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (mikewheaton) [reply packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.styles.ts]: Unfortunately DocumentCard isn't in the toolkit at all, so I've just been matching the previous styles here.

**Comment by dzearing** [comment]
@aftab-hassan I am curious why so many files are changing just 1 or 3 bytes. Is there some sort of unpredictability here with size auditor, or webpack, or is there some EOF character that is sporadically emitting?

**Comment by dzearing** [comment]
@mikewheaton The DocumentCard went up 1K in minified size (which maybe could equate to a few hundred bytes gzipped). I'm curious why; was the javascript styling larger than the legacy styling? Or, was there some sort of extra logic added?

@aditima @aftab-hassan We are all on the same page here already, but just another example of how important drilling into bundle size difference is. If you think in the viewpoint of the author (Mike), you'll be curious about why this was; what kind of info would be valuable? (we talked about diffing the code which could help, but really just glancing at per module breakdown (this module added X bytes, this one removed X bytes) would get you a quicker answer.)

There's also a few other things that made it difficult to parse the numbers:

Change: 1498... Is this bytes? is it minified or gzipped? Is this small, medium or large? I don't have good context to know the impact.

Suggestion:
1. List out a unit. E.g. if less than 1024, print out bytes: "42 bytes"
2. Else print out KB: "1.02 KB"

I don't think you need MB or beyond. :)

Also print both minified and gzipped. Or, short term, change the column header to say "minified" just to be clear we're not taking into account gzip optimization.

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.styles.ts [hunk 15]
```diff
@@ -0,0 +1,109 @@
+import { getGlobalClassNames, FontSizes, FontWeights } from '../../Styling';
+import { IDocumentCardActivityStyleProps, IDocumentCardActivityStyles } from './DocumentCardActivity.types';
+
+const VERTICAL_PADDING = 8;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.styles.ts]: Just for future reference, these consts would be "tokens" to allow them to be configurable.

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.styles.ts [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+
+        selectors: {
+          '&:after': {
+            content: '" "',
+            position: 'absolute',
+            left: '-1px',
+            top: '-1px',
+            right: '-1px',
+            bottom: '-1px',
+            border: `2px solid ${palette.white}`, // Match the background of the card
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.styles.ts]: can't this be inherit, or transparent?

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.styles.ts [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+    details: [
+      classNames.details,
+      {
+        left: multiplePeople
+          ? `${HORIZONTAL_PADDING + IMAGE_SIZE * 1.5 + PERSONA_TEXT_GUTTER}px`
+          : `${HORIZONTAL_PADDING + IMAGE_SIZE + PERSONA_TEXT_GUTTER}px`,
+        height: IMAGE_SIZE,
+        position: 'absolute',
+        top: VERTICAL_PADDING,
+        width: `calc(100% - #{HORIZONTAL_PADDING + IMAGE_SIZE + PERSONA_TEXT_GUTTER + HORIZONTAL_PADDING}px)`
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.styles.ts]: is there any way to avoid this calc?

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.styles.ts]: Just a suggestion. There are a handful of things in css I don't trust that can create performance issues and it's really, really difficult to find them. Things like calc and nested flexboxes and sibling selectors (+) can create layout thrashing.

Comment (mikewheaton) [reply packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.styles.ts]: It should be possible to use flexbox here, but I didn't want to introduce any major refactoring during the conversion.

**Comment by mikewheaton** [comment]
@dzearing I'm not sure why the bundle size went up. There are a lot of new files, and the style objects are slightly larger than CSS, so it may be a case of a lot of small changes adding up.

@Vitalius1 Thanks for all the feedback! I've made the requested changes, could you take another look?

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.base.tsx [hunk 18]
```diff
@@ -32,7 +41,7 @@ export class DocumentCard extends BaseComponent<IDocumentCardProps, any> impleme
     }
 
     // if this element is actionable it should have an aria role
-    const role = this.props.role || (actionable ? (onClick ? 'button' : 'link') : undefined);
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.base.tsx]: Do we need to deprecate this prop now that it's unused?

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.types.ts [hunk 19]
```diff
@@ -63,257 +48,61 @@ export interface IDocumentCardProps extends IBaseProps<IDocumentCard> {
...
-   * Describes DocumentCard status icon.
-   */
-  statusIcon?: string;
-
-  /**
-   * Describe status information. Required field.
-   */
-  status: string;
+export interface IDocumentCardStyles {
+  root: IStyle;
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.types.ts]: nit: Would be nice to have the interface itself to be documented and each section with a short description of what it represents. I just hate to see adding new undocumented APIs.

Same goes for all components 🙄

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.styles.ts [hunk 20]
```diff
@@ -0,0 +1,64 @@
+import { getGlobalClassNames, FontSizes } from '../../Styling';
+import { IDocumentCardActionsStyleProps, IDocumentCardActionsStyles } from './DocumentCardActions.types';
+
+const actionSize = 34;
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.styles.ts]: Capitalize the consts here too.

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.tsx [hunk 21]
```diff
@@ -1,33 +0,0 @@
-import * as React from 'react';
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.tsx]: Really weird why the git history worked on the `DocumentCard` rename to base but here it just removed the file(((

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.types.ts [hunk 22]
```diff
@@ -0,0 +1,57 @@
...
+   */
+  theme?: ITheme;
+
+  /**
+   * Optional override class name
+   */
+  className?: string;
+}
+
+export interface IDocumentCardActionsStyleProps {
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.types.ts]: Document the interface too maybe (not a blocker) 😄

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.types.ts [hunk 23]
```diff
@@ -0,0 +1,57 @@
...
+   * Optional override class name
+   */
+  className?: string;
+}
+
+export interface IDocumentCardActionsStyles {
+  root: IStyle;
+  action: IStyle;
+  views: IStyle;
+  viewsIcon: IStyle;
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.types.ts]: Document all areas if possible 😄  (not a blocker)

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.base.tsx [hunk 24]
```diff
@@ -1,37 +1,47 @@
 import * as React from 'react';
-import { BaseComponent, css } from '../../Utilities';
-import { IDocumentCardActivityProps, IDocumentCardActivityPerson } from './DocumentCard.types';
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActivity.base.tsx]: See in here the renaming preserved the history and you get a nice diff. I wonder why it didn't work on `DocumentCardActions` 🤔

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.base.tsx [hunk 25]
```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { BaseComponent, classNamesFunction } from '../../Utilities';
+import { Icon } from '../../Icon';
+import { IDocumentCardActionsProps, IDocumentCardActionsStyleProps, IDocumentCardActionsStyles } from './DocumentCardActions.types';
+import { IconButton } from '../../Button';
+import { IProcessedStyleSet } from '../../Styling';
+
+const getClassNames = classNamesFunction<IDocumentCardActionsStyleProps, IDocumentCardActionsStyles>();
+
+export class DocumentCardActionsBase extends BaseComponent<IDocumentCardActionsProps, any> {
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.base.tsx]: nit: Seems an overkill to have this a class component vs a functional component

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardDetails.base.tsx [hunk 26]
```diff
@@ -0,0 +1,22 @@
...
+
+  public render(): JSX.Element {
+    const { children, styles, theme, className } = this.props;
+
+    this._classNames = getClassNames(styles!, {
+      theme: theme!,
+      className
+    });
+
+    return <div className={this._classNames.root}>{children}</div>;
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardDetails.base.tsx]: nit: Maybe consider a functional component? (not a blocker)

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardDetails.styles.ts [hunk 27]
```diff
@@ -0,0 +1,27 @@
+import { getGlobalClassNames } from '../../Styling';
+import { IDocumentCardStatusStyleProps, IDocumentCardStatusStyles } from './DocumentCardStatus.types';
+
+const GlobalClassNames = {
+  root: 'ms-DocumentCardDetails'
+};
+
+export const getStyles = (props: IDocumentCardStatusStyleProps): IDocumentCardStatusStyles => {
+  const { className, theme } = props;
+  // const { palette, fonts } = theme;
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardDetails.styles.ts]: Remove comment.

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardDetails.ts [hunk 28]
```diff
@@ -0,0 +1,14 @@
+// Right now there's a class in a bunch of the examples
+// that was never converted into a component. Make a
+// component for it.
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardDetails.ts]: Do we need this comment?

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardLocation.base.tsx [hunk 29]
```diff
@@ -0,0 +1,25 @@
...
+
+  public render(): JSX.Element {
+    const { location, locationHref, ariaLabel, onClick, styles, theme, className } = this.props;
+
+    this._classNames = getClassNames(styles!, {
+      theme: theme!,
+      className
+    });
+
+    return (
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardLocation.base.tsx]: Same... maybe a functional component? (not a blocker)

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardLogo.base.tsx [hunk 30]
```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import { Icon } from '../../Icon';
+import { IProcessedStyleSet } from '../../Styling';
+import { BaseComponent, classNamesFunction } from '../../Utilities';
+import { IDocumentCardLogoProps, IDocumentCardLogoStyleProps, IDocumentCardLogoStyles } from './DocumentCardLogo.types';
+
+const getClassNames = classNamesFunction<IDocumentCardLogoStyleProps, IDocumentCardLogoStyles>();
+
+export class DocumentCardLogoBase extends BaseComponent<IDocumentCardLogoProps, any> {
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardLogo.base.tsx]: Same... maybe functional component (not a blocker)

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardPreview.styles.ts [hunk 31]
```diff
@@ -0,0 +1,83 @@
+import { getGlobalClassNames, FontSizes } from '../../Styling';
+import { IDocumentCardPreviewStyleProps, IDocumentCardPreviewStyles } from './DocumentCardPreview.types';
+
+export const GlobalClassNames = {
+  root: 'ms-DocumentCardPreview',
+  icon: 'ms-DocumentCardPreview-icon',
+  iconContainer: 'ms-DocumentCardPreview-iconContainer',
+  previewFileTypeIcon: 'ms-DocumentCardPreview-icon'
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardPreview.styles.ts]: Does this suppose to have the same global className as on line 6? Maybe leave just `GlobalClassNames.icon` and use it in both places?

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardPreview.styles.ts [hunk 32]
```diff
@@ -0,0 +1,83 @@
...
+    ],
+    fileList: {
+      padding: '16px 16px 0 16px',
+      listStyleType: 'none',
+      margin: 0,
+      selectors: {
+        li: {
+          height: '16px',
+          lineHeight: '16px',
+          marginBottom: '8px',
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardPreview.styles.ts]: nit: again not a blocker but all these values can be just numbers. No need to specify `px` everywhere. Same goes for all the `.styles` files.

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardStatus.base.tsx [hunk 33]
```diff
@@ -0,0 +1,34 @@
...
+    this._classNames = getClassNames(styles!, {
+      theme: theme!,
+      className
+    });
+
+    return (
+      <div className={this._classNames.root}>
+        {statusIcon && <Icon {...iconProps} />}
+        {status}
+      </div>
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardStatus.base.tsx]: Maybe functional component (please don't hate me 🙄 😄 )

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.base.tsx (hunk 18 above)
Comment (mikewheaton) [reply packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.base.tsx]: Which prop is deprecated?

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.base.tsx (hunk 25 above)
Comment (mikewheaton) [reply packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardActions.base.tsx]: Looking around the code base I'm seeing class components everywhere, so I'll leave these as is.

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardPreview.styles.ts (hunk 31 above)
Comment (mikewheaton) [reply packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardPreview.styles.ts]: Thanks for catching that, it should have a different global class name.

**Comment by mikewheaton** [comment]
@yiminwu, could you review this PR?

**Comment by aftab-hassan** [comment]
> @aditima @aftab-hassan We are all on the same page here already, but just another example of how important drilling into bundle size difference is. If you think in the viewpoint of the author (Mike), you'll be curious about why this was; what kind of info would be valuable? (we talked about diffing the code which could help, but really just glancing at per module breakdown (this module added X bytes, this one removed X bytes) would get you a quicker answer.)

+1 We spoke offline as well and are on the same page about drilling down by module. That would help the author to know which specific files brought about a change in `module`, which in turn affected the `chunk`, and ultimately the `asset`.

> There's also a few other things that made it difficult to parse the numbers:
> 
> Change: 1498... Is this bytes? is it minified or gzipped? Is this small, medium or large? I don't have good context to know the impact.

This is bytes minified. David suggested an alternate way of reporting, which we will roll out in the next LightRail PR.

> Also print both minified and gzipped. Or, short term, change the column header to say "minified" just to be clear we're not taking into account gzip optimization.

We'll do gzipped as well.

**Pull request merged**